src/signals/incident-management/containers/IncidentOverviewPage/saga.js

```diff
@@ -61,6 +62,32 @@ export function* fetchIncidents() {
   }
 }
 
+export function* searchIncidents(action) {
+  const { payload } = action;
+
+  try {
+    yield put(applyFilterRefreshStop());
+
+    const incidents = yield call(authCall, CONFIGURATION.SEARCH_ENDPOINT, {
+      q: payload,
+    });
+
+    yield put(requestIncidentsSuccess(incidents));
+  } catch (error) {
+    if (
+      error.response &&
+      error.response.status === 500
+    ) {
+      // Getting an error response with status code 500 from the search endpoint
+      // means that the Elasticsearch index is very likely corrupted. In that
+      // case we simulate a success response without results.
+      yield put(requestIncidentsSuccess({ count: 0, results: [] }));
```

Comment:
Does this mean we yield a success first and then an error on line 87?

Reply:
Yes, for administrative purposes. So that, when we're debugging we can still see that the request was erroneous. The user shouldn't see an error.